source/common/tls/context_config_impl.cc

```diff
@@ -363,23 +376,37 @@ const std::string ClientContextConfigImpl::DEFAULT_CURVES =
 #endif
     "P-256";
 
+absl::StatusOr<std::unique_ptr<ClientContextConfigImpl>> ClientContextConfigImpl::create(
+    const envoy::extensions::transport_sockets::tls::v3::UpstreamTlsContext& config,
+    Server::Configuration::TransportSocketFactoryContext& secret_provider_context) {
+  absl::Status creation_status = absl::OkStatus();
+  std::unique_ptr<ClientContextConfigImpl> ret = absl::WrapUnique(
```

Comment:
`WrapUnique` is a best practice way of getting a `unique_ptr` from an explicit call to `new`. Basically it helps you say things like, "new is forbidden in the codebase unless it's used in WrapUnique since we're sure that can't leak".
https://abseil.io/tips/126

That being said, in this context auto seem fine for the type of `ret` since the type is totally obvious from the RHS.